src/foam/box/socket/SocketServer.js

```diff
@@ -78,11 +78,24 @@ foam.CLASS({
                   while ( true ) {
                     Socket client = serverSocket.accept();
                     client.setSoTimeout(getSoTimeout());
-                    agency.submit(
-                      x,
-                      new SocketServerProcessor(getX(), client),
-                      client.getRemoteSocketAddress().toString()
-                    );
+                    
+                    final X x_ = x;
+                    Thread thread = new Thread(new Runnable() {
```

Comment:
Please setDaemon(true) if not already the default for the thread. 
This will ensure the thread does not block server shutdown. 
Here and above. Thanks.

Reply:
ok

Reply:
Sorry, this is backwards. 
Want the 'accept' to be explicit thread, and then when accepted, pass that to the agency.